coda/coda_mdstore/tests/test_views.py

```diff
@@ -0,0 +1,890 @@
+        response = views.app_node(request, node.node_name)
+        tree = objectify.fromstring(response.content)
+
+        assert len(tree.content.node) == 1
+        assert tree.content.node.name == node.node_name
+        assert tree.content.node.capacity == node.node_capacity
+        assert tree.content.node.path == node.node_path
+        assert tree.content.node.url == node.node_url
+
+    def test_get_request_with_identifier_raises_exception(self, rf):
+        request = rf.get('/')
+        response = views.app_node(request, '001')
+        assert response.status_code == 404
+
+    @mock.patch('coda_mdstore.views.createNode')
+    def test_post_request(self, mock_createNode, rf):
+        node = NodeFactory.create()
+        mock_createNode.return_value = node
+
+        request = rf.post('/', HTTP_HOST='example.com')
+        response = views.app_node(request)
+
+        assert response.status_code == 201
+        assert response['Content-Type'] == 'application/atom+xml'
+        assert response['Location'] == (
+            'http://example.com/APP/node/{0}/'.format(node.node_name)
+        )
+
+    @mock.patch('coda_mdstore.views.updateNode')
+    def test_put_request(self, mock_updateNode, rf):
+        node = NodeFactory.create()
+        mock_updateNode.return_value = node
+
+        request = rf.put('/', HTTP_HOST='example.com')
+        response = views.app_node(request, node.node_name)
+
+        assert response.status_code == 200
+        assert response['Content-Type'] == 'application/atom+xml'
+
+    def test_delete_request(self, rf):
+        node = NodeFactory.create()
+        request = rf.delete('/', HTTP_HOST='example.com')
+        response = views.app_node(request, node.node_name)
+        assert response.status_code == 200
+
+    def test_delete_request_returns_not_found(self, rf):
+        request = rf.delete('/', HTTP_HOST='example.com')
+        response = views.app_node(request, 'DNE-001')
+        assert response.status_code == 404
+
+
+class TestAppBag:
+    """
+    Tests for coda_mdstore.views.app_bag.
+    """
+
+    def test_get_request_returns_not_found(self, rf):
+        request = rf.get('/', HTTP_HOST='example.com')
+        response = views.app_bag(request, 'ark:/00002/id')
+        assert response.status_code == 404
+
+    def test_get_request(self, rf):
+        FullBagFactory.create_batch(10)
+
+        request = rf.get('/', HTTP_HOST='example.com')
+        response = views.app_bag(request)
+
+        assert response.status_code == 200
+        assert response['Content-Type'] == 'application/atom+xml'
+
+    def test_get_request_response_content(self, rf):
+        FullBagFactory.create_batch(10)
+
+        request = rf.get('/', HTTP_HOST='example.com')
+        response = views.app_bag(request)
+
+        tree = objectify.fromstring(response.content)
+        assert len(tree.entry) == 10
+
+    @pytest.mark.xfail(reason="Refactor required.")
+    def test_get_request_with_identifier(self):
```

Comment:
I know there is some wonky code in app_bag for a GET request with identifier, but can you either put a test in for a general GET with identifier request in addition to a test marked to fail for parts that will fail, of if you cannot test any path through this, please put a more specific reason.